Makefile

```diff
@@ -124,10 +124,14 @@ install-extensions:
 build-jupyterlab:
 	jupyter lab build
 
-install-server: lint-server build-server install-server-package # Install backend
+prepare-server:
+	pip install --upgrade pip wheel
+
+only-install-server: prepare-server build-server install-server-package
+
+install-server: lint-server only-install-server # Install backend
```

Comment:
do we really need to start breaking things to this granularity? 
linting on the server is 13 seconds, and the build runs in the background...

Reply:
it's helpful to have linting separate because if it fails it won't fail all part of ci only the linting section

Reply:
we don't need to get extremely granular, but I think that linting, unit tests, and integration tests should all have their own "step" and they shouldn't be duplicated across "jobs"